### PR TITLE
Feat/improved search

### PR DESCRIPTION
**Pull request by justinvdk**
Makes search pane more dynamic. 
Since metadata is still a class which is based on the sheet we received, and the new SearchPane could in theory filter on any categorical data, two helper methods are implemented.

### CODE REVIEW

**Comment by wouterjsmit** [comment]
Looks good, please fix the conflicts.

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java [hunk 1]
```diff
@@ -25,19 +24,24 @@
...
 import java.util.Collection;
-import java.util.HashSet;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.ResourceBundle;
-import java.util.Set;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 @SuppressWarnings("AbbreviationAsWordInName")
```
Comment (cbilstra) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java:54]: Javadoc for class?

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java [hunk 2]
```diff
@@ -95,6 +88,25 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
     initializeExtendedSearchPane();
   }
```
Comment (cbilstra) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java:105]: Javadoc

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java [hunk 3]
```diff
@@ -162,68 +184,173 @@ private void initializeTable() {
...
+      case  "streptomycin":
+        return metaData.streptomycin;
+      case  "digitalSpoligotype":
+        return metaData.digitalSpoligotype;
+      case  "sex":
+        return metaData.sex.name();
+      default:
+        throw new RuntimeException("Undefined column in temp method");
+    }
   }
```
Comment (cbilstra) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java:339]: Javadoc

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java (hunk 1 above)
Comment (justinvdk) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java:54]: Added

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java (hunk 2 above)
Comment (justinvdk) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/SearchPaneController.java:105]: Added

**Comment by wouterjsmit** [comment]
Please fix your conflicts

**Comment by justinvdk** [comment]
Rebased on latest dev.

**Pull request merged**